base_attachment_object_storage/models/ir_attachment.py

```diff
@@ -68,109 +68,114 @@ def _register_hook(self):
         if update_module:
             self.env['ir.attachment'].sudo()._force_storage_to_object_storage()
 
-    @api.model
-    def _save_in_db_domain(self):
+    @property
+    def _object_storage_default_force_db_config(self):
+        return {"image/": 50000, "application/javascript": 0, "text/css": 0}
+
+    def _get_storage_force_db_config(self):
+        param = self.env['ir.config_parameter'].sudo().get_param(
+            'ir_attachment.storage.force.database',
+        )
+        storage_config = None
+        if param:
+            try:
+                storage_config = const_eval(param)
+            except (SyntaxError, TypeError, ValueError):
+                _logger.exception(
+                    "Could not parse system parameter"
+                    " 'ir_attachment.storage.force.database', reverting to the"
+                    " default configuration.")
+
+        if not storage_config:
+            storage_config = self._object_storage_default_force_db_config
+        return storage_config
+
+    def _store_in_db_instead_of_object_storage_domain(self):
         """Return a domain for attachments that must be forced to DB
 
-        Read the docstring of ``_save_in_db_anyway`` for more details.
+        Read the docstring of ``_store_in_db_instead_of_object_storage`` for
+        more details.
+
+        Used in ``force_storage_to_db_for_special_fields`` to find records
+        to move from the object storage to the database.
 
         The domain must be inline with the conditions in
-        ``_save_in_db_anyway``.
+        ``_store_in_db_instead_of_object_storage``.
         """
-        excluded_model_settings = self.env['ir.config_parameter'].sudo().\
-            get_param('excluded.models.storedb', default='')
-        excluded_model_for_db_store = excluded_model_settings.split(',')
-        mimetypes_settings = self.env['ir.config_parameter'].sudo().get_param(
-            'mimetypes.list.storedb', default='')
-        mimetypes_for_db_store = mimetypes_settings.split(',')
-        filesize = self.env['ir.config_parameter'].sudo().get_param(
-            'file.maxsize.storedb', default='0')
-        domain = [
-            '|',
-            # assets are stored in 'ir.ui.view'
-            ('res_model', '=', 'ir.ui.view'),
-            '&', '&',
-            ('file_size', '<', int(filesize)),
-            ('res_model', 'not in', excluded_model_for_db_store),
-        ]
-        domain += ['|'] * (len(mimetypes_for_db_store) - 1)
-        domain += [('mimetype', '=like', mimetype) for mimetype in
-                   mimetypes_for_db_store]
+        domain = []
+        storage_config = self._get_storage_force_db_config()
+        for mimetype_key, limit in storage_config.items():
+            part = [("mimetype", "=like", "{}%".format(mimetype_key))]
+            if limit:
+                part = AND([part, [("file_size", "<=", limit)]])
```

Comment:
Here I guess, we must ensure limit is a Integer > 0, but it's probably better to do it `_get_storage_force_db_config` ?

Reply:
If someone puts a negative limit, no file will match this mimetype, which means they stay on the object storage, which is equivalent to not put the mimetype in the configuration. It doesn't really make sense to use a negative limit, but it does no harm, so I don't think we should add code to handle this.